src/storage/heap_file.c

```diff
@@ -24863,11 +24848,52 @@ heap_get_visible_version (THREAD_ENTRY * thread_p, const OID * oid, OID * class_
 */
 SCAN_CODE
 heap_scan_get_visible_version (THREAD_ENTRY * thread_p, const OID * oid, OID * class_oid, RECDES * recdes,
-			       HEAP_SCANCACHE * scan_cache, int ispeeking, int old_chn)
+			       RECDES * peeked_recdes, HEAP_SCANCACHE * scan_cache, int ispeeking, int old_chn)
 {
   SCAN_CODE scan = S_SUCCESS;
   HEAP_GET_CONTEXT context;
 
+  /*
+   * The process below should be within heap_get_visible_version_internal(), 
+   * but it's an added shortcut for performance improvement. Under certain specific conditions, 
+   * it allows for skipping the process of initializing and cleaning the context and the 
+   * heap_get_visible_version_internal() function. This brings the current CUBRID's heap scan 
+   * performance closer to the heap scan performance of CUBRID before the introduction of MVCC. 
+   * Following is the explanation for the code below.
+   * Before fetching a record, check peeked_recdes to see if the record type is REC_HOME,
+   * and it's being PEEKed (meaning there's no need to COPY the record data to a new space). 
+   * In this case, we can use peeked_recdes as the record without executing the
+   * heap_get_visible_version_internal() function. If the conditions above are not met,
+   * or the mvcc_snapshot does not satisfy, then carry out the necessary steps through 
+   * the heap_get_visible_version_internal() function.
+   */
+  if (peeked_recdes->type == REC_HOME && ispeeking == PEEK)
```

Comment:
I think peeked_recdes should be checked to see if it is null.
Since heap_scan_get_visible_version is only called from heap_next_internal, the following code may be sufficient.
```
assert (peeked_recdes != NULL);
```